File: sol/3542/3542.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0
        st = [0]
        for x in nums:
            while st[-1] > x:
                res += 1
                st.pop()
            if st[-1] != x:
                st.append(x)
        res += len(st) - 1
        return res

File: sol/3542/3542.cpp
struct Solution {
    int minOperations(vector<int>& nums) {
        int res = 0;
        vector<int> st(1, 0);
        for (int x : nums){
            while (st.back() > x){
                ++res;
                st.pop_back();
            }
            if (st.back() != x){
                st.push_back(x);
            }
        }
        res += st.size() - 1;
        return res;
    }
};
